.buildkite/filter-pipeline.py

```diff
@@ -70,24 +70,34 @@ def dfs(name, depends_on_dict, visited):
             dfs(adjacent, depends_on_dict, visited)
 
 
-def build_tree(steps):
+def build_key_to_label(steps):
+    key_to_label = {}
+    for step in steps:
+        if "key" in step and "label" in step:
+            key_to_label[step["key"]] = step["label"]
+    return key_to_label
```

Comment:
I'm not sure I understand the need for this, can you explain?

Given that this is based on buildkite's `depends_on`, which in turn uses `key`s, shouldn't the logic of this script change to use `key`s as well (and not use `label`s at all)?

Reply:
only steps that used in `depends_on` have to have `key` attribute, while we still want to visit all reverse dependencies as well, which don't have `key`, thus it's easier to navigate through dependency tree using `label`

Reply:
Oh I see, got it.

Can you please add a comment to this function saying why we need it then?